middleware/http/transport_test.go

```diff
@@ -206,21 +206,21 @@ func TestTransportRequestSamplerOverridesSamplingFromContext(t *testing.T) {
 		}
 
 		sp := tracer.StartSpan("op1")
-		defer sp.Finish()
 		ctx := zipkin.NewContext(context.Background(), sp)
 
 		req, _ := http.NewRequest("GET", srv.URL, nil)
-		transport, _ := NewTransport(
+		tr, _ := NewTransport(
 			tracer,
 			TransportRequestSampler(c.RequestSampler),
 		)
 
-		_, err = transport.RoundTrip(req.WithContext(ctx))
+		_, err = tr.RoundTrip(req.WithContext(ctx))
+		sp.Finish()
```

Comment:
interesting no more defer.. I guess it is defer ordering issue ya?

Reply:
resource leak as the defer is in a loop